helix-term/src/commands.rs

```diff
@@ -1970,8 +1924,10 @@ fn global_search(cx: &mut Context) {
     let current_path = doc_mut!(cx.editor).path().cloned();
 
     let show_picker = async move {
-        let all_matches: Vec<FileResult> =
-            UnboundedReceiverStream::new(all_matches_rx).collect().await;
+        let all_matches: Vec<FileResult> = UnboundedReceiverStream::new(all_matches_rx)
+            .map(|(path, _, line_num)| FileResult::new(&path, line_num as usize - 1))
```

Comment:
Seems more wasteful to do this upfront rather than calculating it for matches only (after search_path)

Reply:
can you guiding me how to write a common function or sink that accepts a transform lambda? I am stuck there, so I send all three fields to the channel.